snapcraft/project/_project.py

```diff
@@ -57,6 +58,7 @@ def __init__(
         self._is_managed_host = is_managed_host
         self._project_dir = project_dir
         self._work_dir = work_dir
+        self._project_dir_hash = self._get_project_directory_hash()
```

Comment:
Why not just call the method when needed?

Reply:
Certainly could, the cost is inconsequential anyhow.